internal/cloudapi/v2/server.go

```diff
@@ -445,22 +446,79 @@ func (s *Server) enqueueKojiCompose(taskID uint64, server, name, version, releas
-	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	// prepared to become a config variable
+	const depsolveTimeout = 5
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*depsolveTimeout)
 	defer cancel()
 
-	// wait until job is in a pending state
-	var token uuid.UUID
+	jobResult := &worker.ManifestJobByIDResult{
+		Manifest: nil,
+		ManifestInfo: worker.ManifestInfo{
+			OSBuildComposerVersion: common.BuildVersion(),
+		},
+	}
+
 	var dynArgs []json.RawMessage
 	var err error
+	token := uuid.Nil
 	logWithId := logrus.WithField("jobId", manifestJobID)
+
+	defer func() {
+		// token == uuid.Nil indicates that no worker even started processing
+		if token == uuid.Nil {
+			if jobResult.JobError != nil {
+				// set all jobs to "failed"
+				// osbuild job will fail as dependency
+				jobs := map[string]uuid.UUID{
+					"depsolve":         depsolveJobID,
+					"containerResolve": containerResolveJobID,
+					"ostreeResolve":    ostreeResolveJobID,
+					"manifest":         manifestJobID,
+				}
+
+				for jobName, jobID := range jobs {
+					if jobID != uuid.Nil {
+						err := workers.SetFailed(jobID, jobResult.JobError)
+						if err != nil {
+							logWithId.Errorf("Error failing %s job: %v", jobName, err)
+						}
+					}
+				}
+
+			} else {
+				logWithId.Errorf("Internal error, no worker started depsolve but we didn't get a reason.")
+			}
+		} else {
+			result, err := json.Marshal(jobResult)
+			if err != nil {
+				logWithId.Errorf("Error marshalling manifest job results: %v", err)
+			}
+			err = workers.FinishJob(token, result)
+			if err != nil {
+				logWithId.Errorf("Error finishing manifest job: %v", err)
+			}
+			if jobResult.JobError != nil {
+				logWithId.Errorf("Error in manifest job %v: %v", jobResult.JobError.Reason, err)
+			}
+		}
+	}()
+
+	// wait until job is in a pending state
 	for {
 		_, token, _, _, dynArgs, err = workers.RequestJobById(ctx, "", manifestJobID)
-		if err == jobqueue.ErrNotPending {
+		if errors.Is(err, jobqueue.ErrNotPending) {
 			logWithId.Debug("Manifest job not pending, waiting for depsolve job to finish")
 			time.Sleep(time.Millisecond * 50)
 			select {
 			case <-ctx.Done():
-				logWithId.Warning("Manifest job dependencies took longer than 5 minutes to finish, or the server is shutting down, returning to avoid dangling routines")
+				logWithId.Warning(fmt.Sprintf("Manifest job dependencies took longer than %d minutes to finish,"+
+					" or the server is shutting down, returning to avoid dangling routines", depsolveTimeout))
+
+				jobResult.JobError = clienterrors.New(clienterrors.ErrorDepsolveTimeout,
+					"Timeout while waiting for package dependency resolution",
+					"There may be a temporary issue with compute resources. "+
+						"We’re looking into it, please try again later.",
```

Comment:
Not sure about the error message here, plus the alignment looks off to me? why is the `We're` more indented than the rest.

Also we don't tend to say that we're looking into it, people should be encouraged to open tickets when necessary. I'd just keep the first line really. Also technically it's not just package dependency, it's all resolve jobs.

We could ask the user to make sure all included sources are reachable and responsive etc…

Reply:
I removed the last line for now, but ` please try again later.` is at least some hint what to do - I'm not sure if it's good to instruct them to contact support or leave them without a help "what to do"